File: csharp/perfect-numbers/PerfectNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
      if (number <= 0)
      {
        throw new ArgumentOutOfRangeException("number must be greater than zero");
      }
        var factors = FindFactors(number);
        var aliquotSum = factors.Sum();
       
        if (aliquotSum == number) 
        {
          return Classification.Perfect;
        }
        if (aliquotSum > number)
        {
          return Classification.Abundant;
        }
        return Classification.Deficient;
    }
    private static IEnumerable<int> FindFactors(int number)
    {
      var factors = new List<int>();
      for (var x=1; x < number; x++)
      {
          if (number % x == 0)
          {
            factors.Add(x);
          }
      }
      return factors;
    }
    
}

File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    if number <= 0:
        raise ValueError("number must be greater than zero.")
    
    s = sum(find_factors(number))
    
    if s == number:
        return "perfect"
    if s >= number:
        return "abundant"
    if s <= number:
        return "deficient"

def find_factors(number):
    factors = []
    for x in range(1, number):
        if number % x == 0:
            factors.append(x)
        x += 1
    return factors